Makefile

```diff
@@ -3,37 +3,45 @@
 # make sure we turn on go modules
 export GO111MODULE := on
 
-# for building out the customd app
-BUILDOUT ?= customd 
-BUILD_VERSION ?= $(shell git describe --tags)
-BUILD_FLAGS := -mod=readonly -ldflags "-X github.com/iov-one/weave.Version=$(BUILD_VERSION)"
+TOOLS := cmd/customd cmd/customcli
 
 # MODE=count records heat map in test coverage
 # MODE=set just records which lines were hit by one test
 MODE ?= set
 
+# Check if linter exists
+LINT := $(shell command -v golangci-lint 2> /dev/null)
+
 # for dockerized prototool
 USER := $(shell id -u):$(shell id -g)
 DOCKER_BASE := docker run --rm --user=${USER} -v $(shell pwd):/work iov1/prototool:v0.2.2
 PROTOTOOL := $(DOCKER_BASE) prototool
 PROTOC := $(DOCKER_BASE) protoc
 WEAVEDIR=$(shell go list -m -f '{{.Dir}}' github.com/iov-one/weave)
 
-all: clean test import-spec install
-
-clean:
-	rm -f ${BUILDOUT}
+all: import-spec test lint
 
 build:
 	go build $(BUILD_FLAGS) -o $(BUILDOUT) ./cmd/customd
 
+dist:
+	cd cmd/bnsd && $(MAKE) dist
+
 install:
-	go install $(BUILD_FLAGS) ./cmd/customd
+	for ex in $(TOOLS); do cd $$ex && make install && cd -; done
 
 test:
+	@# customd binary is required by some tests. In order to not skip them, ensure customd binary is provided and in the latest version.
+	#go install -mod=readonly ./cmd/customd
+
 	go vet -mod=readonly ./...
 	go test -mod=readonly -race ./...
 
+lint:
+	@go mod vendor
```

Comment:
You are vendoring because you want to lint all dependencies or because the dependencies source code is required for proper linting?

Reply:
I got this code weave untouched.  Apparently, the reason is the latter. When I run `make lint` without `go mod vendor`, I get many`go build github.com/...: no Go files in` error.